### PR TITLE
Fix for some issues (installer & client)

### PR DESCRIPTION
**Pull request by floriangoeldi**
Can you review my changes? :)

Commit (1) comment:
- Added a force quit option to the app, if this is set, the application will quit. Otherwise it'll just hide. (#89)
- Fixed bug in windows, where you had to triple-click the tray icon show the app (#77)

Commit (2) comment:
- Added check, if the application is running, because this lead to
errors
- Added option to add a shortcut to the autostart folder (#57)
- Added option to launch the app after the installation (#37)
- Added option to choose the installation directory (#41)
- Added the shortcuts as options, so you can choose, if you want a
desktop, autostart or start menu shortcut (a part of #96)
- Enabled / Added these pages: components, browse installation folder,
finish page
- Extended the comments, so it's easier to find the correct spots

### CODE REVIEW

File: resources/windows/installer.nsi [hunk 1]
```diff
@@ -52,27 +56,56 @@ Caption "${productName} Setup"
...
 Function .onInit
 
     ; Extract banner image for welcome page
     InitPluginsDir
     ReserveFile "${banner}"
     File /oname=$PLUGINSDIR\banner.bmp "${banner}"
+    
+    ; Check if the application is currently running, show message if it is
+    retryInstallation:
+    FindWindow $0 "Chrome_WidgetWin_1" "Rocket.Chat"
```
Comment (rodrigok) [inline resources/windows/installer.nsi]: Is there some way to get the application name `Rocket.Chat` from parameter `productName`?

File: resources/windows/installer.nsi [hunk 2]
```diff
@@ -84,9 +117,15 @@ Function welcome
 
 FunctionEnd
 
-; Installation declarations
-Section "Install"
+;   --------------------------------
+;    Installation sections
+;   --------------------------------
+
+Section "Rocket.Chat Client"
```
Comment (rodrigok) [inline resources/windows/installer.nsi]: Can we get the application name from parameter here too?

File: resources/windows/installer.nsi [hunk 3]
```diff
@@ -122,13 +211,40 @@ UninstallCaption "Uninstall ${productName}"
...
-; Custom uninstall confirm page
+;   --------------------------------
+;    Uninstaller init
+;   --------------------------------
+
+Function un.onInit
+
+    ; Check if the application is currently running, show message if it is
+    retryUninstall:
+    FindWindow $0 "Chrome_WidgetWin_1" "Rocket.Chat"
```
Comment (rodrigok) [inline resources/windows/installer.nsi]: Can we use parameter here too?

**Comment by floriangoeldi** [comment]
I'll try and test it later in the evening. If it works I'll commit it to the request.
Otherwise I'll get back to you! ;)

**Comment by rodrigok** [comment]
Nice, then we can ship this within the new version 1.1.0

**Comment by floriangoeldi** [comment]
So I've replaced the remaining `Rocket.Chat` occurrences with the variable.

Important is, that the window name of the Client always has to be the same as the product name variable, so the installer can find the open windows with `FindWindow`

**OffTopic:**
Sorry for the merge commit, which is also included in this pull request. That this doesn't happen again, do I understand this correctly: If I have something like a feature branch in my fork, the merges of your develop branch will be into my develop branch, so in the pull request from the branch they won't be included?

**Comment by rodrigok** [comment]
@floriangoeldi awesome.

About your merge question, yes, but as you will open the PR to the develop branch is better to merge from our develop into your feature branch first to prevent unmergeable PRs (conflicts).

So, you can use `rebase` instead `merge` if you want to prevent the merge commits ;)

**Pull request merged**